File: lib/JFFUtils/JFFUtilsTest/NumberOfCharacterFromStringTest.m
#import "NumberOfCharacterFromStringTest.h"

@implementation NumberOfCharacterFromStringTest

-(void)testNumberOfCharacterFromString
{
    XCTAssertEqual((NSUInteger)0, [@"" numberOfCharacterFromString: @""]);
    
    XCTAssertEqual((NSUInteger)0, [@"" numberOfCharacterFromString: @"1"]);
    
    XCTAssertEqual((NSUInteger)2, [@"11" numberOfCharacterFromString: @"1"]);
    
    XCTAssertEqual((NSUInteger)2, [@"21212" numberOfCharacterFromString: @"1"]);
    
    XCTAssertEqual((NSUInteger)5, [@"00021212000" numberOfCharacterFromString: @"21"]);
    
    XCTAssertEqual((NSUInteger)7, [@"00032123120000" numberOfCharacterFromString: @"213"]);
}

-(void)testNumberOfStringsFromString
{
    XCTAssertEqual((NSUInteger)3, [@"aaa" numberOfStringsFromString: @"a" ]);
    
    XCTAssertEqual((NSUInteger)1, [@"aaa" numberOfStringsFromString: @"aa" ]);
    
    XCTAssertEqual((NSUInteger)1, [@"ab a" numberOfStringsFromString: @"ab" ]);
    
    XCTAssertEqual((NSUInteger)1, [@"a abc" numberOfStringsFromString: @"abc" ]);
    
    XCTAssertEqual((NSUInteger)0, [@"a ab c" numberOfStringsFromString: @"abc" ]);
    
    XCTAssertEqual((NSUInteger)3, [@"ababab" numberOfStringsFromString: @"ab" ]);
}

@end

File: iAsync.Swift/lib/JUtils/JUtilsTests/NumberOfCharacterFromStringTest.swift
//
//  NumberOfCharacterFromStringTest.swift
//  JUtilsTests
//
//  Created by Vladimir Gorbenko on 14.06.14.
//  Copyright (c) 2014 EmbeddedSources. All rights reserved.
//

import XCTest

class NumberOfCharacterFromStringTest: XCTestCase {
    
    func testNumberOfCharacterFromString() {
        
        XCTAssertEqual(0, "".numberOfCharacterFromString(""))
        
        XCTAssertEqual(0, "".numberOfCharacterFromString("1"))
        
        XCTAssertEqual(2, "11".numberOfCharacterFromString("1"))
        
        XCTAssertEqual(2, "21212".numberOfCharacterFromString("1"))
        
        XCTAssertEqual(5, "00021212000".numberOfCharacterFromString("21"))
        
        XCTAssertEqual(7, "00032123120000".numberOfCharacterFromString("213"))
    }
    
    func testNumberOfStringsFromString() {
        
        XCTAssertEqual(3, "aaa".numberOfStringsFromString("a"))
        
        XCTAssertEqual(1, "aaa".numberOfStringsFromString("aa"))
        
        XCTAssertEqual(1, "ab a".numberOfStringsFromString("ab"))
        
        XCTAssertEqual(1, "a abc".numberOfStringsFromString("abc"))
        
        XCTAssertEqual(0, "a ab c".numberOfStringsFromString("abc"))
        
        XCTAssertEqual(3, "ababab".numberOfStringsFromString("ab"))
    }
}
